### PR TITLE
Fix asm2f.test_simd15

### PR DESCRIPTION
**Pull request by jgravelle-google**
With #6801 we no longer serialize all current settings we want to pass. This uncovered a bug: before for `-Oz` tests, the test runner's Settings would have `INLINING_LIMIT=0`, and we would forward this to our invocation of emcc. `emcc.py`'s `parse_args` has the behavior that `-Oz` (and `-Os`) overwrite the `INLINING_LIMIT` setting, which we used to later clobber with `-s INLINING_LIMIT=0`.
So, by no longer forwarding every bit of Settings state, we're actually testing `-Oz`'s behavior, which has the inlining limit. And that causes test_simd15 to fail.

The fix here is: `emcc test_simd15.c -Oz -msse -msse2` didn't used to work outside the test runner, it shouldn't work in the test runner, so skip it.

### CODE REVIEW

File: tests/test_core.py [hunk 1]
```diff
@@ -7845,6 +7846,7 @@ def setUp(self):
 
     assert emcc_args is not None
     self.emcc_args = emcc_args[:]
+    Settings.load(self.emcc_args)
```
Comment (kripken) [inline tests/test_core.py]: why is this needed now? is it part of the fix, or unrelated?

File: tests/test_core.py [hunk 2]
```diff
@@ -7859,12 +7861,6 @@ def setUp(self):
       if arg.startswith('-O'):
         Building.COMPILER_TEST_OPTS.append(arg) # so bitcode is optimized too, this is for cpp to ll
 
-    # forward  -s K=V
-    for i in range(len(self.emcc_args)):
-      if self.emcc_args[i] == '-s':
-        key, value = self.emcc_args[i + 1].split('=', 1)
-        self.settings_mods[key] = json.loads(value)
-
```
Comment (kripken) [inline tests/test_core.py]: i'm guessing this code here was unneeded before and this is just cleanup?

File: tests/test_core.py (hunk 1 above)
Comment (kripken) [reply tests/test_core.py]: I see, thanks. lgtm!

Comment (sbc100) [reply tests/test_core.py]: Are you sure this is correct?  The old code seems to populate the settings_mods, which gets passed to emcc subproccess on the command line.  This new code seems to modify the test runners in-memory copy of the Settings only.

Comment (sbc100) [reply tests/test_core.py]: Oh I see, it doesn't matter since those emcc args will be passed to emcc subprocess on the command line anyway.  

LGTM.
